docs/components/index.md

```diff
@@ -0,0 +1,315 @@
+
+```csharp
+@using System
+@implements IDisposable
+
+...
+
+@functions {
+    public void Dispose()
+    {
+        ...
+    }
+}
+```
+
+## Render child content
+
+Blazor components can render child content with an element that matches the child component's name. The following Index page renders the content of the `HeadingComponent` (*HeadingComponent.cshtml*):
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/Index.cshtml?start=1&end=11)]
+
+A child component can receive property assignments from their calling parent. In the following example, the `ParentComponent` sets the value of the `Title` property in the `ChildComponent`:
+
+*ParentComponent.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/ParentComponent.cshtml?start=1&end=7)]
+
+*ChildComponent.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/ChildComponent.cshtml)]
+
+The body of the Bootstrap-styled panel is provided by `ChildContent`. `ChildContent` is bound to the `<ChildComponent>` element's content in the parent component by convention.
+
+## Routing
+
+Routing in Blazor is achieved by providing a route template to each accessible component in the app. Route templates are converted into [RouteAttribute](/dotnet/api/microsoft.aspnetcore.mvc.routeattribute) implementations. Multiple route templates can be applied to a component. The following component responds to requests for `/BlazorRoute` and `/DifferentBlazorRoute`:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/BlazorRoute.cshtml?start=1&end=4)]
+
+## Route parameters
+
+Blazor components can receive and use route parameters from the route template provided in the `@page` directive:
+
+*RouteParameter.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/RouteParameter.cshtml?start=1&end=8)]
+
+Not all route parameter constraints are supported at this time. Optional parameters aren't currently supported, so two `@page` directives are applied in the example above. The first permits navigation to the component without a parameter. The second `@page` directive takes the `{text}` route parameter and assigns the value to the `Text` property.
+
+For more information on attribute routing, see [Routing](https://docs.microsoft.com/aspnet/core/fundamentals/routing).
+
+## JavaScript/TypeScript interop
+
+To call JavaScript libraries or custom JavaScript/TypeScript code from .NET, the current approach is to register a named function with JavaScript/TypeScript. Place the `registerFunction` call in the *index.html* file or a JavaScript file (*\*.js*) loaded by the *index.html* file. Place the inline JavaScript or `<script>` tag below `<script type="blazor-boot"></script>`, and the JavaScript/TypeScript loads at the correct time and only executes once. 
+
+```javascript
+Blazor.registerFunction('doPrompt', message => {
```

Comment:
For wider browser compatibility, can we avoid the lambda syntax? i.e., replace with:

```
Blazor.registerFunction('doPrompt', function(message) {
    return prompt(message);
});
```